pkg/cli/cli.go

```diff
@@ -47,14 +52,61 @@ func ParseConfig(configFile string) (vsphere.Config, error) {
 	return cfg, err
 }
 
-// TODO (fanz) : Perform vSphere configuration health check on VM:
-func CheckVSphereConfig(config vsphere.Config) error {
+// CheckVSphereConfig performs vSphere health check on VMs
+// TODO (fanz) : support checking network
+func CheckVSphereConfig(ctx context.Context, o *ClientOption) error {
+	c, err := o.GetClient()
+	if err != nil {
+		return err
+	}
+	vc := view.NewManager(c)
+	cv, err := vc.CreateContainerView(ctx, c.ServiceContent.RootFolder, []string{"VirtualMachine"}, true)
+	if err != nil {
+		return err
+	}
+	defer cv.Destroy(ctx)
+	var vms []mo.VirtualMachine
+	var vm *object.VirtualMachine
+	var config extraConfig
+	config.Set("disk.enableUUID=1")
+
+	f := find.NewFinder(c, true)
+	err = cv.Retrieve(ctx, []string{"VirtualMachine"}, []string{"summary"}, &vms)
+	if err != nil {
+		return err
+	}
+	for _, v := range vms {
+		if v.Summary.Config.Uuid == "" {
+			name := v.Summary.Config.Name
+			vm, err = f.VirtualMachine(ctx, name)
```

Comment:
No need to use `Finder` here, instead you can use: `object.NewVirtualMachine(c, v.Reference())`